Zend/zend_compile.c

```diff
@@ -6350,11 +6358,12 @@ static uint32_t count_match_conds(zend_ast_list *arms)
 
 	for (uint32_t i = 0; i < arms->children; i++) {
 		zend_ast *arm_ast = arms->child[i];
-		if (arm_ast->child[0] == NULL) {
+		zend_ast_list *conds = zend_ast_get_list(arm_ast->child[0]);
+
+		if (conds->child[0]->kind == ZEND_AST_DEFAULT) {
```

Comment:
You'll need to either prevent `match (1) { 2, default => ... }` or `default` in all conditions.

Reply:
Wouldn't preventing `default` be a BC break?

Reply:
`match (1) { 2, default => ... }` is illegal syntax before this RFC. You're currently only handling `default` as the only condition in the match arm.

Reply:
I understand I'm only checking the first condition, but I didn't realise it was illegal to combine default with other conditions before. In that case, I think we should keep it that way (deny it), but either way, we have to traverse all the children, do we not? Could probably wrap that behaviour in a function, though...

Reply:
Yes, that's what I meant with "check for default in all conditions".